app.ts

```diff
@@ -0,0 +1,411 @@
+					);
+				}
+				if (this.paginationStart === 1) {
+					$(".prev").css({ display: "none" });
+				} else {
+					$(".prev").css({ display: "block" });
+				}
+			})
+			.catch(error => {
+				throw new Error(`Failed rendering the pagination: ${error}`);
+			});
+	}
+
+	/** Handles all the functionality related to pagination. */
+	initializePagination(): void {
+		// Render the specific amount of records to the DOM based on the current page that it gets from the 
+		// element's attribute value. 
+		$(".pagination").on("click", ".pagination-item", (event) => {
+			$('.pagination-item').prop('disabled', true);
+			return this.apiManager.getRecordCount()
+				.then(count => {
+					$('.search-input').val('');
+					let returnedId = <string>($(event.target).attr("value"));
+					this.currentPage = parseInt(returnedId);
+					const maxRecords = this.recordsPerPage();
+					let fromID = Math.ceil(this.currentPage * maxRecords - (maxRecords - 1));
+					if (this.difference > 0 && this.difference < maxRecords) {
+						fromID = fromID + this.difference;
+					}
+					let toID = fromID + (maxRecords - 1);
+					if (fromID > count + 1 || toID > count + 1) {
+						toID = count;
+						fromID = toID - maxRecords;
+						this.currentFromID = fromID;
+					}
+					this.currentFromID = fromID;
+					$(".pagination-item").removeClass("active");
+					$(event.target).addClass("active");
+					const self = this;
+					$(".pagination-item").each(function () {
+						let elementID = <string>($(this).attr('value'));
+						let currentPageString = self.currentPage.toString();
+						if (elementID == currentPageString) {
+							$(this).addClass('active');
+						}
+					});
+					return this.showRecords(fromID, toID)
+						.then(() => {
+							$('.pagination-item').prop('disabled', false)
+						});
+				})
+				.catch(error => {
+					throw new Error(`Failed showing the records when clicking on the page button: ${error}`);
+				});
+		})
+
+		/** Gives the next set of page numbers based on the last page on the pagination. */
+		$(".next").on("click", () => {
+			this.paginationStart = this.paginationEnd + 1;
+			this.paginationEnd = this.paginationStart + 9;
+			$(".pagination").empty();
+			this.pageNumbers(this.paginationStart, this.paginationEnd);
+		});
+
+		/** Gives the previous set of pages numbers based on the last page in the pagination. */
+		$(".prev").on("click", () => {
+			this.paginationEnd = this.paginationStart - 1;
+			this.paginationStart = this.paginationEnd - 9;
+			$(".pagination").empty();
+			this.pageNumbers(this.paginationStart, this.paginationEnd);
+		});
+	}
+
+	/** Handles all the functionality related to the search. */
+	initializeSearch(): void {
+		/** Prevents certain characters to be entered in the input field. */
+		$('.search-input').on('keydown', (e) => {
+			if (e.key === 'e' || e.key === 'E' || e.key === '.' || e.key === '+' || e.key === '*' || e.key === '-') {
```

Comment:
I would recommend that you make use of Regex to only allow digits instead of a long if statement for all these separate case-by-case checks.